src/main/java/org/apache/commons/imaging/formats/tiff/datareaders/ImageDataReader.java

```diff
@@ -518,8 +519,71 @@ protected void applyPredictorToBlock(final int width, final int height, final in
         return samples;
     }
 
+
+    protected int[] unpackIntSamples(
+        final int width,
+        final int height,
+        final int scansize,
+        final byte[] bytes,
+        final int predictor,
+        final int bitsPerSample,
+        final ByteOrder byteOrder)
+        throws ImageReadException {
+        final int bytesPerSample = bitsPerSample / 8;
+        final int nBytes = bytesPerSample * scansize * height;
+        final int length = bytes.length < nBytes ? nBytes / scansize : height;
+
+        final int[] samples = new int[scansize * height];
```

Comment:
Could this be explored by a malicious user to cause a high-memory usage in the code?

Reply:
That's a concern....  A malicious user could create a toxic TIFF file.  An unlucky developer could do it accidentally. More likely, there would be a perfectly valid TIFF file that is just too big.  I recently encountered a file that was 65536-by-65536 pixels in size...   The problem is definition what is "too big".    Back in the day, a 100 megapixel image was problematic.  On a modern computer (or mobile phone), it is commonplace.

Any application code that is going to reach this point will have already accessed a TiffDictionary structure and can know the width and height of the image (or sub-image) that it is accessing.  So I think that we have to leave that particular issue up to the application code.